server/odc-service/src/main/java/com/oceanbase/odc/service/common/util/WebRequestUtils.java

```diff
@@ -118,6 +118,37 @@ public static String getClientAddress(HttpServletRequest request) {
         return request.getRemoteAddr();
     }
 
+    public static String getTrulyClientIp(HttpServletRequest request) {
+        return getTrulyClientIp(getClientAddress(request));
+    }
+
+    /**
+     * <pre>
+     * Find truly ip of client.
+     * The X-Forwarded-For header may contain multiple IP addresses, separated
+     * by commas, and typically, the first non-unknown IP is considered to be the client's IP address.
+     * </pre>
+     * 
+     * @author keyang.lk
+     * @date 2024-07-02
+     * @param remoteAddress
+     * @return The truly ip of remoteAddress
+     */
+    public static String getTrulyClientIp(String remoteAddress) {
```

Comment:
your method may be renamed to `getTrulyClientIpFromXForwardedForHttpHeader` if this method can only process the remote address from this http header~

Reply:
Considering that the client's real IP address can be obtained not only from the X-Forwarded-For header but also potentially from other headers such as Remote_ADDR, would it be appropriate to rename this method to getTrulyClientIpFromHttpHeader?

Reply:
how about `getFirstClientIpAddress`, and may move this method to audit module.

considering:
1. we cannot confirm this method will return the 'truly' client ip
2. it is not a problem in web request area